families-api/app/main.py

```diff
@@ -0,0 +1,225 @@
+from typing import Generic, Optional, TypeVar
+
+from fastapi import APIRouter, Depends, FastAPI
+from pydantic_settings import BaseSettings
+from sqlalchemy import create_engine
+from sqlmodel import Field, Relationship, Session, SQLModel, func, select
+
+
+class FamilyCorpusLink(SQLModel, table=True):
+    __tablename__ = "family_corpus"  # type: ignore[assignment]
+    corpus_import_id: int | None = Field(
+        default=None, foreign_key="corpus.import_id", primary_key=True
+    )
+    family_import_id: int | None = Field(
+        default=None, foreign_key="family.import_id", primary_key=True
+    )
+
+
+class Corpus(SQLModel, table=True):
+    __tablename__ = "corpus"  # type: ignore[assignment]
+    import_id: str | None = Field(default=None, primary_key=True)
+    title: str
+    families: list["Family"] = Relationship(
+        back_populates="corpus", link_model=FamilyCorpusLink
+    )
+
+
+class FamilyGeographyLink(SQLModel, table=True):
+    __tablename__ = "family_geography"  # type: ignore[assignment]
+    geography_id: int | None = Field(
+        default=None, foreign_key="geography.id", primary_key=True
+    )
+    family_import_id: int | None = Field(
+        default=None, foreign_key="family.import_id", primary_key=True
+    )
+
+
+class GeographyBase(SQLModel):
+    id: int | None = Field(default=None, primary_key=True)
+    display_value: str
+    value: str
+    type: str
+    slug: str
+
+
+class Geography(GeographyBase, table=True):
+    __tablename__ = "geography"  # type: ignore[assignment]
+    value: str
+    parent_id: int = Field(foreign_key="geography.id")
+    # the relationship stuff here is a little non-standard
+    # there doesn't seem to be an easier way to do this with SQLModel
```

Comment:
I've no issue with non-standard workarounds, but would be keen to understand if this affects / entails any lazy-loading magic or anything like that? Since we're pushing for simplicity, a good understanding of that would be useful, as many DB inefficiencies I've debugged in my time have had lazy/eager loading ORM logic root causes

Reply:
This is inherited from current RDS setup - I've updated the comment as per, related to [the comment above](https://github.com/climatepolicyradar/navigator-backend/pull/501#discussion_r2088714869).